sdk/python/kfp/dsl/tasks_group.py

```diff
@@ -150,22 +174,151 @@ class Condition(TasksGroup):
       ::
 
         task1 = my_component1(...)
-        with Condition(task1.output=='pizza', 'pizza-condition'):
+        with dsl.If(task1.output=='pizza', 'pizza-condition'):
             task2 = my_component2(...)
     """
 
     def __init__(
         self,
-        condition: pipeline_channel.ConditionOperator,
+        condition,
```

Comment:
nit: type annotation?

Reply:
I intentionally removed this, since I personally find it a bit confusing and I suspect users do too. See the [current reference docs](https://kubeflow-pipelines.readthedocs.io/en/master/source/dsl.html#kfp.dsl.Condition) which use this annotation.

While annotation is technically correct, it isn't informative to the user and exposes an [implementation detail](https://github.com/kubeflow/pipelines/blob/760c1589edbe58bbd77611222a66a17b371a0d08/sdk/python/kfp/dsl/pipeline_channel.py#L151-L167). We don't actually want the user to provide a `ConditionOperator` (name changed in this PR)... we want them to provide a comparative expression involving one or more pipeline channels (i.e., to think of it as a "future" boolean).

WDYT?